wsl-pro-service/internal/system/landscape.go

```diff
@@ -75,14 +77,24 @@ func (s *System) fixAndEnableLandscapeFromConfig(ctx context.Context, landscapeC
 	}
 
 	// TODO: check foreground/background
-	cmd := s.backend.LandscapeConfigExecutable(ctx, "--config", landscapeConfigPath, "--silent")
+	cmd := s.backend.LandscapeConfigExecutable(ctx, "--config", landscapeConfigPath, "--silent", "--register-if-needed")
 	if _, err := runCommand(cmd); err != nil {
 		return fmt.Errorf("could not enable Landscape: %v", err)
 	}
 
 	return nil
 }
 
+func areConfigsEqual(a, b string) bool {
+	fieldsFunc := func(r rune) bool {
+		return unicode.IsSpace(r) || r == '='
```

Comment:
This is a very niche edge case, but if a URL (or any other config value) contains an `=`, then the config wouldn't be correctly reregistered if the `=` was subsequently removed/added. This might not be a valid scenario in this case, but just a thought

Reply:
I'm not aware of that possibility, but let's not take the risk. The functions doing the changes can and should report whether anything changed or not. So let's leverage that and avoid string comparisons at all.